src/sentry/sentry_apps/api/bases/sentryapps.py

```diff
@@ -131,7 +128,12 @@ def handle_exception_with_details(self, request, exc, handler_context=None, scop
 
     def _handle_sentry_app_exception(self, exception: Exception):
         if isinstance(exception, SentryAppIntegratorError) or isinstance(exception, SentryAppError):
-            response = Response({"detail": str(exception)}, status=exception.status_code)
+            response_body: dict[str, Any] = {"detail": exception.message}
+            public_context = exception.public_context
+            if public_context:
```

Comment:
nit: you can use the walrus operator

```suggestion
            if public_context := exception.public_context:
```